### DISCUSSION TITLE
Question about GPU support in AP or perspective

### DISCUSSION THREAD

**Discussion by arapiraca** [QutEcoacoustics/audio-analysis#557] [General]
Hi Everebody,

I'm a begginer in AP program, but I have some experience with HPC computing and appreciate a lot the openmp implementation in the code, but, I need run faster and think about GPU support for the code. Searching for the GPU word in the repository I found in [src/AudioAnalysisTools/SURFFeatures.cs] a code that mention a lot of GPU's calls. Is it a isolate experimental code from some developer or is it a code to be launched with GPU support. Are there any perspectives to a GPU base AP release?

Thanks the developers for the great job!
Best Regards
Antonio
![Captura de tela de 2023-07-13 15-33-02](https://github.com/QutEcoacoustics/audio-analysis/assets/13936580/0c8665db-1c30-4dc3-8dd0-d5fd5148a90d)

**Comment by atruskie** [comment]
Hi Antonio,

There is no GPU support. For the types of analysis AP does we believe GPU analysis wouldn't add much benefit for the cost involved with adding support.

That isn't to say AP couldn't be faster. But again, given our available resources, and that AP was faster than some comparable software (we did measure a while ago), so we focused instead on improving features.

Make sure you're using the --parallel flag, and make sure you're using array jobs on your HPC.

Why do you need to run it faster? Which part is slow?

--Anthony

**Comment by arapiraca** [comment]
Hi Anthony,

Thank you for your fast reply. I'm a physicist/theoretical chemistry trying to help a biologist collaborator on his tasks to study acoustic soundscapes in some conservation units/fores/Savanna etc in Minas Gerais, Brazil. My friend used AP on his Phd at New Zeland and now he is studiyng those soundscapes here. I used AP with multithreading functionality on my 6 core laptop anda I get good results. But, the bioacoustic area is more and more big data based and I'm thinking about solutions to help the stuff go faster. The audio files are getting bigger.

In fact, I need try AP on a decent workstation server or cluster to study the scalability of the code increasing threads in a more appropriated computational environment.

I'll do my tests and report here.

Antonio
